lib/src/Uncertainty/Bayesian/RandomWalkMetropolisHastings.cxx

```diff
@@ -98,198 +86,124 @@ RandomWalkMetropolisHastings* RandomWalkMetropolisHastings::clone() const
+Distribution RandomWalkMetropolisHastings::getInstrumental() const
 {
-  for (UnsignedInteger i = 0; i < calibrationStrategy_.getSize(); ++ i)
-  {
-    calibrationStrategy_[i] = calibrationStrategy;
-  }
+  return instrumental_;
 }
 
-RandomWalkMetropolisHastings::CalibrationStrategyCollection RandomWalkMetropolisHastings::getCalibrationStrategyPerComponent() const
+
+void RandomWalkMetropolisHastings::setCalibrationRange(const Interval & calibrationRange)
 {
-  return calibrationStrategy_;
+  if ( calibrationRange.getDimension() != 1 ) throw InvalidDimensionException(HERE) << "Range should be 1-d. Got " << calibrationRange.getDimension();
+  calibrationRange_ = calibrationRange;
 }
 
-void RandomWalkMetropolisHastings::setCalibrationStrategyPerComponent(const CalibrationStrategyCollection& calibrationStrategy)
+Interval RandomWalkMetropolisHastings::getCalibrationRange() const
 {
-  const UnsignedInteger dimension = proposal_.getSize();
-  if(dimension != calibrationStrategy.getSize()) throw InvalidDimensionException(HERE) << "The proposal dimension (" << dimension << ") does not match the calibration strategy size (" << calibrationStrategy.getSize() << ").";
-  calibrationStrategy_ = calibrationStrategy;
+  return calibrationRange_;
 }
 
 
-void RandomWalkMetropolisHastings::setProposal(const RandomWalkMetropolisHastings::DistributionCollection& proposal)
+void RandomWalkMetropolisHastings::setExpansionFactor(Scalar expansionFactor)
 {
-  const UnsignedInteger dimension = getPrior().getDimension();
-  if (proposal.getSize() != dimension) throw InvalidDimensionException(HERE) << "The proposal dimension (" << proposal.getSize() << ") does not match the prior dimension (" << dimension << ").";
+  if (!( expansionFactor > 1.0)) throw InvalidArgumentException(HERE) << "Expansion factor should be > 1. Got " << expansionFactor;
+  expansionFactor_ = expansionFactor;
+}
 
-  for (UnsignedInteger i = 0; i < dimension; ++ i)
-  {
-    Bool symmetric = proposal[i].isElliptical();
-    symmetric = symmetric && (std::abs(proposal[i].getMean()[0]) < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon"));
-    if (!symmetric) throw InvalidArgumentException(HERE) << "The proposal density is not symmetric.";
-  }
-  proposal_ = proposal;
+Scalar RandomWalkMetropolisHastings::getExpansionFactor() const
+{
+  return expansionFactor_;
 }
 
 
-RandomWalkMetropolisHastings::DistributionCollection RandomWalkMetropolisHastings::getProposal() const
+void RandomWalkMetropolisHastings::setShrinkFactor(Scalar shrinkFactor)
 {
-  return proposal_;
+  if ( (shrinkFactor <= 0.0) || (shrinkFactor >= 1.0) ) throw InvalidArgumentException(HERE) << "Shrink factor should be in (0, 1). Got " << shrinkFactor;
+  shrinkFactor_ = shrinkFactor;
 }
 
+Scalar RandomWalkMetropolisHastings::getShrinkFactor() const
+{
+  return shrinkFactor_;
+}
+
+void RandomWalkMetropolisHastings::setCalibrationPeriod(const UnsignedInteger calibrationPeriod)
```

Comment:
I think all "CalibrationXxx" methods and attributes should be renamed "AdaptationXxx". What do you think @MerlinKeller ?

Reply:
I agree!

Reply:
yes, it avoids confusion with the other classes